clickhouse/tests/test_clickhouse.py

```diff
@@ -37,10 +39,32 @@ def test_can_connect(aggregator, instance):
     (It used to be submitted only once on check init, which led to customer seeing "no data" in the UI.)
     """
     check = ClickhouseCheck('clickhouse', {}, [instance])
+
+    # Test for consecutive healthy clickhouse.can_connect statuses
     num_runs = 3
     for _ in range(num_runs):
         check.run()
-    aggregator.assert_service_check("clickhouse.can_connect", count=num_runs)
+    aggregator.assert_service_check("clickhouse.can_connect", count=num_runs, status=check.OK)
+    aggregator.reset()
+
+    # Test 1 healthy connection --> 2 Unhealthy service checks --> 1 healthy connection. Recovered
+    check.run()
+    with mock.patch('clickhouse_driver.Client', side_effect=NetworkError('Connection refused')):
```

Comment:
why do you need two levels of mocks here? is it not sufficient to just use the first mock or does `ping_clickhouse` not depend on that?